### PR TITLE
upload config file for data retention

### LINKED ISSUE #581: Upload configuration file as part of the sample deployment
Parent #578
Dependency #579

**Pre-requirements:**
For this task is necessary to have the original sample deployed so e2e testing is possible.

**Success criteria:**
As part of the bicep deployment, the configuration file needs to be uploaded to the agreed storage place on the right format and with the right permissions.

### PR DESCRIPTION
**Pull request by elenaterenzi**
# Type of PR
- Code changes
- CI-CD changes

## Purpose
Data Retention - add configuration file as part of the deployment steps

## Author pre-publish checklist
- [ ] Added test to prove my fix is effective or new feature works
- [x] No PII in logs
- [ ] Made corresponding changes to the documentation

## Validation steps
- run the deploy.sh script and verify that the configuration file has been uploaded

## Issues Closed or Referenced

- Closes #581

### CODE REVIEW

File: single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/config/data-retention-config.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (ydaponte) [inline single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/config/data-retention-config.json:1]: Can you please rename the file to datalake_config.json and replace with the file I've pasted in the teams chat this morning? The last proposal is also available here: https://github.com/orgs/Azure-Samples/projects/15/views/1?pane=issue&itemId=17285672

File: single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh [hunk 2]
```diff
@@ -77,6 +77,23 @@ if [[ -z $arm_output ]]; then
...
+azure_storage_account=$(echo "$arm_output" | jq -r '.properties.outputs.storage_account_name.value')
+azure_storage_key=$(az storage account keys list \
+    --account-name "$azure_storage_account" \
+    --resource-group "$resource_group_name" \
+    --output json | jq -r '.[0].value')
+
+echo "Uploading Data Retention config file within the file system."
+# Upload Configuration file
+az storage blob upload --container-name 'config' --account-name "$azure_storage_account" --account-key "$azure_storage_key" \
+    --file scripts/config/data-retention-config.json --name "data-retention/config.json" --overwrite
```
Comment (ydaponte) [inline single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh:94]: Rename to be consistent with the previous comment datalake_config.json (in fact there are more configuration aspects beyond the retention). Retention might be even not considered on the configuration file. We can assume 5 years retention overall - we don't need to have it repeated all along the file properties. But we can discuss in the stand-up.

File: single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh [hunk 3]
```diff
@@ -77,6 +77,23 @@ if [[ -z $arm_output ]]; then
     exit 1
 fi
 
+
+##########################################
+# Upload Data Retention Configuration File
```
Comment (ydaponte) [inline single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh:82]: Correct the comment - there are more configurations beyond the data retention

File: single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh (hunk 2 above)
Comment (ydaponte) [inline single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh:94]: data-retention folder can also disappear - we can have the file directly under config

**Review by ydaponte — CHANGES_REQUESTED** [review]
The code is working well, beyond the comments I left there is one thing missing - probably you need to allow the MSI to read the config container and file, because the pipeline will read with the MSI. The MSI name is available as the WS name during the deployment, is just a matter of add it to the container and file (probably you will need to use the ACL CLI that we were discussing on the stand-up. Let me know if you have questions! And thank YOU!

File: single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh (hunk 2 above)
Comment (elenaterenzi) [reply single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh:94]: makes sense I would like to keep data-retention in the file name though as I think in the future this container might contain other kinds of configs

File: single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh (hunk 3 above)
Comment (elenaterenzi) [reply single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh:82]: what other configurations?

File: single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/config/data-retention-config.json (hunk 1 above)
Comment (elenaterenzi) [reply single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/config/data-retention-config.json]: changed accordingly, resolving conversation

File: single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh (hunk 2 above)
Comment (elenaterenzi) [reply single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh]: changed accordingly, resolving conversation

File: single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh (hunk 3 above)
Comment (elenaterenzi) [reply single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh]: changed accordingly, resolving conversation

File: single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh (hunk 2 above)
Comment (elenaterenzi) [reply single_tech_samples/synapseanalytics/sample2_synapseserverless/scripts/deploy_infrastructure.sh]: changed accordingly, resolving conversation

**Comment by elenaterenzi** [comment]
> because the pipeline will read with the M



> The code is working well, beyond the comments I left there is one thing missing - probably you need to allow the MSI to read the config container and file, because the pipeline will read with the MSI. The MSI name is available as the WS name during the deployment, is just a matter of add it to the container and file (probably you will need to use the ACL CLI that we were discussing on the stand-up. Let me know if you have questions! And thank YOU!

There is already a role assignment as Storage Contributor to the Synapse worskpace MSI for the storage account where the config container lives, that should be enough I guess unless you want something else? note that you as a user won't have access to the container if that was your concern, in that case I can grant read access but I guess it would be better if end users don't have access to the config so we remove the possibility that they would not tamper with data lake ACLs?

**Review by ydaponte — APPROVED** [review]

**Pull request merged**